### PR TITLE
add command to create a template function

### PR DESCRIPTION
**Pull request by justin1121**

### CODE REVIEW

File: cmd/cape/cmd/function.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+)
+
+var functionCmd = &cobra.Command{
+	Use:   "function",
```
Comment (gavinuhma) [inline cmd/cape/cmd/function.go:12]: Looks good. I do wonder if this should simply be `cape create`

**Review by gavinuhma — DISMISSED** [review]

File: cmd/cape/cmd/function.go (hunk 1 above)
Comment (justin1121) [reply cmd/cape/cmd/function.go:12]: Hmmmm I think we will have another command under `function`. The one to build dependencies.

```
cape function create ....
cape function build ....
```

Comment (justin1121) [reply cmd/cape/cmd/function.go:12]: But both could just be under `cape` directly.

Comment (gavinuhma) [reply cmd/cape/cmd/function.go:12]: I do think it's nice when you can look at the top level and see the most important stuff.

I feel like build could be part of deploy. Like `cape deploy --build` . Unless we see a reason to build separately from deploy 

Some user feedback has suggested that `cape test` is confusing and would be better as a flag to run.

Comment (justin1121) [reply cmd/cape/cmd/function.go:12]: Good point. I like `cape deploy --build`.

Comment (justin1121) [reply cmd/cape/cmd/function.go]: I just moved it to top level for now. We can always move things around with top level becomes too cluttered.

File: cmd/cape/cmd/function.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+)
+
+var createCmd = &cobra.Command{
+	Use:   "create [name]",
+	Short: "Creates the boilerplate code for creating a function",
```
Comment (bendecoste) [inline cmd/cape/cmd/function.go:13]:
```suggestion
	Short: "Creates the boilerplate code for a function",
```

**Review by bendecoste — CHANGES_REQUESTED** [review]

File: cmd/cape/cmd/function.go (hunk 2 above)
Comment (bendecoste) [reply cmd/cape/cmd/function.go:13]: Or 'create a directory containing a cape function or something like that

File: cmd/cape/cmd/function.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+)
+
+var createCmd = &cobra.Command{
+	Use:   "create [name]",
```
[the remark this answers is not in the document]
Comment (justin1121) [reply cmd/cape/cmd/function.go:17]: Going to stick with where the [original ticket](https://capeprivacy.atlassian.net/browse/CAPE-894?atlOrigin=eyJpIjoiNTU2YWQ5OTA5YmE5NDlmNjg3ODJhYWIyMzM2MDllY2MiLCJwIjoiaiJ9) specified the command to go  We can adjust later.

**Review by bendecoste — APPROVED** [review]

File: cmd/cape/cmd/function.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+		return UserError{Msg: "unable to create app.py file", Err: err}
+	}
+	defer appF.Close()
+
+	_, err = appF.Write([]byte(appContents))
+	if err != nil {
+		return UserError{Msg: "unable to write to app.py", Err: err}
+	}
+
+	reqF, err := os.Create(filepath.Join(name, "requirements.txt"))
```
Comment (rahul-ramesh) [inline cmd/cape/cmd/function.go:75]: Nit: This is actually the requirements.txt file that failed to be created

Comment (rahul-ramesh) [reply cmd/cape/cmd/function.go:75]: Do we want to roll back if one was created, but the other wasn't?

Comment (justin1121) [reply cmd/cape/cmd/function.go:75]: O i'll fix the error messages. Thanks! Atomically creating these files probably wouldn't hurt but isn't super important IMO. Can open out a ticket for now.

**Pull request merged**